frontend/src/features/public-form/components/FormFields/VisibleFormFields.tsx

```diff
@@ -0,0 +1,49 @@
+import { useEffect, useState } from 'react'
+import { Control, FieldValues, useWatch } from 'react-hook-form'
+
+import { FormFieldDto } from '~shared/types/field'
+import { FormColorTheme, LogicDto } from '~shared/types/form'
+
+import { getVisibleFieldIds } from '~features/logic/utils'
+
+import { FieldFactory } from './FieldFactory'
+
+interface VisibleFormFieldsProps {
+  control: Control<FieldValues>
```

Comment:
would it be possible to avoid the repeated validations by passing a more specific type here?

Reply:
i don't think so; since the key is just `fieldId`, which has nothing to do with the field values...